index.d.ts

```diff
@@ -0,0 +1,97 @@
+import { FastifyPluginAsync } from "fastify";
+import { MercuriusPlugin } from "mercurius";
+export interface PolicyFieldOptions {
+  ttl?: number;
+  storage?: MercuriusCacheStorage;
+  extendKey?: Function;
+  skip?: Function;
+  invalidate?: Function;
+  references?: Function;
+}
+
+export type PolicyFieldName = string;
+export type PolicyField = Record<PolicyFieldName, PolicyFieldOptions | object>;
+export type PolicyName = string;
+export type MercuriusCachePolicy = Record<PolicyName, PolicyField>;
+
+export interface MercuriusCacheStorageOptions {
+  client?: object;
+  invalidate?: boolean | { invalidate: boolean; referencesTTL: number };
```

Comment:
invalidation?: boolean | { referencesTTL: number };

btw it's different between memory and redis, so I think a specific option type is better